mbuild/compound.py

```diff
@@ -802,6 +802,19 @@ def remove(self, objs_to_remove):
             if child.contains_rigid:
                 self.root._reorder_rigid_ids()
 
+        # Remove port of removed part
+        for port in self.all_ports():
+            if id(port.anchor) not in [id(i) for i in self.particles()]:
+                self.remove(port)
+
+        # Remove empty subcompound
+        import mbuild as mb
+        if not isinstance(self, mb.Port) and self.name != 'subport':
+            if len(self.children) == 0:
+                print("Working on {}".format(self))
+                self.root.remove(self)
```

Comment:
I think you're running into an infinite recursion error since you keep calling `remove` on `self`.

Reply:
Yeah, I am trying to fix that now